make/migrations/postgresql/0030_1.11.0_schema.up.sql

```diff
@@ -0,0 +1,40 @@
+/* TODO remove the table artifact_2 and use the artifact instead after finishing the upgrade work */
+CREATE TABLE artifact_2
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  /* image, chart, etc */
+  type          varchar(255),
+  media_type    varchar(255),
+  manifest_media_type varchar(255),
+  project_id    int NOT NULL,
+  repository_id int NOT NULL,
+  digest        varchar(255) NOT NULL,
+  size          bigint,  
+  push_time     timestamp default CURRENT_TIMESTAMP,
+  platform      varchar(255),
+  extra_attrs   text,
+  annotations   text,
+  revision      int NOT NULL,
+  CONSTRAINT unique_artifact_2 UNIQUE (repository_id, digest)
+);
+
+CREATE TABLE tag
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  repository_id int NOT NULL,
+  artifact_id   int NOT NULL,
+  name          varchar(255),
+  push_time     timestamp default CURRENT_TIMESTAMP,
+  pull_time     timestamp,
+  revision      int NOT NULL,
+  CONSTRAINT unique_tag UNIQUE (repository_id, name)
+);
+
+/* artifact_reference records the child artifact referenced by parent artifact */
+CREATE TABLE artifact_reference
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  artifact_id   int NOT NULL,
```

Comment:
let's clarify which column is for parent artifact which column point to child.

Reply:
could we call the parent id directly?

Reply:
Updated